src/parse_attr.rs

```diff
@@ -1,63 +1,64 @@
+use macaddr::MacAddr;
 use std::convert::TryInto;
 
-/// Parse a vec of bytes as hex String
-pub fn parse_hex(input: &Vec<u8>) -> String {
-    let value: Vec<char> = hex::encode_upper(input).chars().collect();
-    let split = value
-        .chunks(2)
-        .map(|chunk| chunk.iter().collect::<String>())
-        .collect::<Vec<String>>();
-
-    split.join(":")
-}
-
 /// Parse a vec of bytes as a String
-pub fn parse_string(input: &Vec<u8>) -> String {
-    String::from_utf8_lossy(&input).to_owned().to_string()
+pub fn parse_string(input: &[u8]) -> String {
+    String::from_utf8_lossy(input)
+        .trim_matches(char::from(0))
+        .to_string()
 }
 
-/// Parse a vec of bytes as u8
-pub fn parse_u8(input: &Vec<u8>) -> u8 {
-    let to_array =
-        |slice: &[u8]| -> [u8; 1] { slice.try_into().expect("slice with incorrect length") };
-
-    u8::from_le_bytes(to_array(input))
+/// Parse a vec of bytes as a mac address
+pub fn parse_macaddr(input: &[u8]) -> MacAddr {
+    if input.len() == 6 {
+        let array: [u8; 6] = input
+            .try_into()
+            .expect("Slice with incorrect number of bytes");
+        array.into()
+    } else if input.len() == 8 {
+        let array: [u8; 8] = input
+            .try_into()
+            .expect("Slice with incorrect number of bytes");
+        array.into()
+    } else {
+        panic!("Mac address should be 6 or 8 bytes");
+    }
 }
 
 /// Parse a vec of bytes as i8
-pub fn parse_i8(input: &Vec<u8>) -> i8 {
+pub fn parse_i8(input: &[u8]) -> i8 {
     let to_array =
         |slice: &[u8]| -> [u8; 1] { slice.try_into().expect("slice with incorrect length") };
 
     i8::from_le_bytes(to_array(input))
 }
 
 /// Parse a vec of bytes as u16
-pub fn parse_u16(input: &Vec<u8>) -> u16 {
+pub fn parse_u16(input: &[u8]) -> u16 {
     let to_array =
         |slice: &[u8]| -> [u8; 2] { slice.try_into().expect("slice with incorrect length") };
 
     u16::from_le_bytes(to_array(input))
 }
 
 /// Parse a vec of bytes as u32
-pub fn parse_u32(input: &Vec<u8>) -> u32 {
+pub fn parse_u32(input: &[u8]) -> u32 {
     let to_array =
         |slice: &[u8]| -> [u8; 4] { slice.try_into().expect("slice with incorrect length") };
 
     u32::from_le_bytes(to_array(input))
 }
 
 /// Parse a vec of bytes as i32
-pub fn parse_i32(input: &Vec<u8>) -> i32 {
+pub fn parse_i32(input: &[u8]) -> i32 {
     let to_array =
         |slice: &[u8]| -> [u8; 4] { slice.try_into().expect("slice with incorrect length") };
 
     i32::from_le_bytes(to_array(input))
 }
 
 /// Parse a vec of bytes as u64
-pub fn parse_u64(input: &Vec<u8>) -> u64 {
+pub fn parse_u64(input: &[u8]) -> u64 {
```

Comment:
Seems like it would be worthwhile to pull in the byteorder crate and use the `ReadBytesExt` trait from there directly where you need these parse functions.